services/data/src/react/hooks/useDataQuery.ts

```diff
@@ -30,24 +37,39 @@ export const useDataQuery = (
         lazy: initialLazy = false,
     }: QueryOptions = {}
 ): QueryRenderInput => {
-    const variablesHash = useRef<string | null>(null)
-    const [variables, setVariables] = useState(initialVariables)
-    const [enabled, setEnabled] = useState(!initialLazy)
     const [staticQuery] = useStaticInput<Query>(query, {
         warn: true,
         name: 'query',
     })
+    const [variablesUpdateCount, setVariablesUpdateCount] = useState(0)
```

Comment:
Not sure I love this name - technically there's one case where it's incorrect (if the query is lazy and you then call `refetch()` without any variables)

Reply:
`forceUpdate`?

```js
const useForceUpdate = () => {
  const [, setCount] = useState(0)
  return () => setCount(count => count + 1)
}

const useDataQuery = () => {
  const forceUpdate = useForceUpdate()
  // ...
  
            // This promise does not currently reject on errors
            const refetchPromise = new Promise(resolve => {
                queryState.current.refetchCallback = data => {
                    resolve(data)
                }
            })

            // Trigger a react-query refetch
            forceUpdate()

            return refetchPromise
}
```

Reply:
Yeah that's a better name... we're technically including the refetch / variable update count as a React DevTools debug value, though I don't know how much value it actually provides

Reply:
Sidebar... we could expose the update count from the same hook.  This could very well be overkill:

```js
const useForceUpdate = () => {
  const [count, setCount] = useState(0)
  const forceUpdate = () => setCount(count => count + 1)
  Object.defineProperty(forceUpdate, "count", {
      get : function () {
          return count
      }
  })
}

const useDataQuery = () => {
  const forceUpdate = useForceUpdate()
  // ...
  
            // This promise does not currently reject on errors
            const refetchPromise = new Promise(resolve => {
                queryState.current.refetchCallback = data => {
                    resolve(data)
                }
            })
            
            // Set count as debug value
            useDebugValue(forceUpdate.count)

            // Trigger a react-query refetch
            forceUpdate()

            return refetchPromise
}
```

Reply:
^ I think this is definitely overkill

Reply:
Yeah I think we can introduce this when we need it. For now I think I'd stick to the forceUpdate pattern @mediremi suggested. That seems nice to me.